### DISCUSSION TITLE
mapControl how to limit the zoom level

### DISCUSSION THREAD

**Discussion by zimuyanghua** [Mapsui/Mapsui#2728] [Q&A]
`Avalonia` doesn't seem to have this attribute
![QQ_1722931151401](https://github.com/user-attachments/assets/bccee1ff-6221-43d7-a66e-e7493ddb4e33)

**Comment by guldud** [comment]
I would like to know as well. It's odd that this is not a part of any samples. The behavior whereby the map shrinks to a centered point is so different from the experience we all have on daily basis using Apple or Google products. So it really stands out.

**Comment by guldud** [comment]
In Avalonia, something like the below resizes the map when resizing the window, and makes sure you cannot zoom out beyond the actual space occupied by the map control. 

But, you can still drag the map edges to the center. Does anyone have an idea how to disallow this?

 ```
if (this.FindControl<ContentControl>("MapContentControl") is {} mapContentControl)
        {
            mapContentControl.Content = mapControl;
            
            mapContentControl.SizeChanged += (s, e) =>
            {
                var navigator = map.Navigator;
                
                navigator.ZoomToPanBounds();
                navigator.OverrideZoomBounds = new MMinMax(navigator.Resolutions[19], navigator.Viewport.Resolution);
            };
        }
```

**Comment by guldud** [comment]
Took a peek again and using this line alone

 map.Navigator.Limiter = new ViewportLimiterKeepWithinExtent();

fixes zooming, panning and so on.  It's even in the doc's. Sigh.

**Comment by pauldendulk** [comment]
This has changed in v5 and can now be done by assigning Map.Navigator.ZoomBounds.
